app/src/main/java/com/example/traders/watchlist/allCrypto/AllCryptoFragment.kt

```diff
@@ -0,0 +1,47 @@
+package com.example.traders.watchlist.allCrypto
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import com.example.traders.databinding.FragmentTabAllCryptoBinding
+import com.example.traders.watchlist.adapters.WatchListAdapter
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+class AllCryptoFragment : Fragment() {
+    private val viewModel: AllCryptoViewModel by viewModels()
+    private var adapter: WatchListAdapter? = null
+
+    override fun onCreateView(
+            inflater: LayoutInflater, container: ViewGroup?,
+            savedInstanceState: Bundle?
+    ): View? {
+        val binding = FragmentTabAllCryptoBinding.inflate(inflater, container, false)
+        viewModel.addItemsToList()
```

Comment:
this is not really the best place to use ViewModel functions and create adapters. Because view is not created yet here. However, since we don't have binding as a global variable it's okay to leave it like this for now. We should figure out how to access viewBinding in everywhere.